lms/djangoapps/discussion/rest_api/api.py

```diff
@@ -913,8 +918,13 @@ def create_comment(request, comment_data):
         raise ValidationError({"thread_id": ["This field is required."]})
     cc_thread, context = _get_thread_and_context(request, thread_id)
 
+    course = context["course"]
+    user = context["cc_requester"]
+    if is_course_discussion_in_blackout(course) and not has_discussion_privileges(user, course.id):
```

Comment:
Let's have a utility method called
```
def discussion_closed_for_user(user, course):
    return is_course_discussion_in_blackout(course) and not has_discussion_privileges(user, course.id)
```

once that is done, we can simply use it here.
```
if discussion_closed_for_user(course, user):
    raise DiscussionBlackedOutException
```

b p : 
0 0 : 0
0 1: 0
1 0: 1
1 1: 0